<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="dysbiosisR">
<title>Introduction • dysbiosisR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="dysbiosisR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dysbiosisR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Introduction.html">Introduction</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/microsud/dysbiosisR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/microsud/dysbiosisR/blob/HEAD/vignettes/Introduction.Rmd" class="external-link"><code>vignettes/Introduction.Rmd</code></a></small>
      <div class="d-none name"><code>Introduction.Rmd</code></div>
    </div>

    
    
<p>In human microbiome research, there is interest in measuring
“dysbiosis” which in many cases relies on comparing a case group with a
control (without known disease). Several methods have been reported in
literature and reviewed by Wei S et al., in their 2021 article <a href="https://doi.org/10.1128/AEM.00395-21" class="external-link">Determining gut microbial
dysbiosis: a review of applied indexes for assessment of intestinal
microbiota imbalances</a>. It is important to note that the term
‘dysbiosis’ is a vaguely defined term in most instances and different
methods are employed. We noticed a lack of tools for convenient and
reproducible measurement of “dysbiosis” measures/scores/index. To
address this gap we accumulated some commonly used approaches into a
more standardized manner in an R package we call
<code>dysbiosisR</code>.<br>
We would also like to direct the readers and users to debates
surrounding the concept of dysbiosis and suggest to interpret these
measures by considering its pitfalls.</p>
<ol style="list-style-type: decimal">
<li><p><a href="https://sfamjournals.onlinelibrary.wiley.com/doi/full/10.1111/1751-7915.13479" class="external-link">Problems
with the concept of gut microbiota dysbiosis</a></p></li>
<li><p><a href="https://doi.org/10.1128%2FmBio.01492-17" class="external-link">Dysbiosis and
Its Discontents</a></p></li>
<li><p><a href="https://doi.org/10.1038/nmicrobiol.2016.228" class="external-link">Dysbiosis
is not an answer</a></p></li>
</ol>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microsud/dysbiosisR" class="external-link">dysbiosisR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: microbiome</span></span>
<span><span class="co">#&gt; Loading required package: phyloseq</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; microbiome R package (microbiome.github.com)</span></span>
<span><span class="co">#&gt;     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  Copyright (C) 2011-2022 Leo Lahti, </span></span>
<span><span class="co">#&gt;     Sudarshan Shetty et al. &lt;microbiome.github.io&gt;</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'microbiome'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ggplot2':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     alpha</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     transform</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"WirbelJ_2018"</span><span class="op">)</span></span></code></pre></div>
<p>Here we use the test data from WirbelJ et al., (2018). The data
consists of CRC and controls samples.</p>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">WirbelJ_2018</span></span>
<span><span class="co">#&gt; phyloseq-class experiment-level object</span></span>
<span><span class="co">#&gt; otu_table()   OTU Table:         [ 530 taxa and 125 samples ]</span></span>
<span><span class="co">#&gt; sample_data() Sample Data:       [ 125 samples by 137 sample variables ]</span></span>
<span><span class="co">#&gt; tax_table()   Taxonomy Table:    [ 530 taxa by 6 taxonomic ranks ]</span></span>
<span><span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 530 tips and 529 internal nodes ]</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">WirbelJ_2018</span><span class="op">)</span><span class="op">$</span><span class="va">disease</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     CRC healthy </span></span>
<span><span class="co">#&gt;      60      65</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dysbiosis-measures">Dysbiosis Measures<a class="anchor" aria-label="anchor" href="#dysbiosis-measures"></a>
</h2>
<div class="section level3">
<h3 id="dysbiosis-score-based-on-median-community-level-variation">Dysbiosis score based on median community level variation<a class="anchor" aria-label="anchor" href="#dysbiosis-score-based-on-median-community-level-variation"></a>
</h3>
<p>Calculates median variation in a given sample compared to a reference
sample group. Here will calculate median variation for a sample compared
to a reference sample set. The user can provide a custom distance
matrix. If the user provides a Bray-Curtis dissimilarity matrix, then
the resulting score is comparable to the ‘dysbiosis score’ reported in
<a href="https://www.nature.com/articles/s41586-019-1237-9" class="external-link">Lloyd-Price
J, Arze C, Ananthakrishnan AN et al. (2019)</a></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist.mat</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/distance.html" class="external-link">distance</a></span><span class="op">(</span><span class="va">WirbelJ_2018</span>, <span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="va">ref.samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/meta.html" class="external-link">meta</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html" class="external-link">subset_samples</a></span><span class="op">(</span><span class="va">WirbelJ_2018</span>, <span class="va">disease</span> <span class="op">==</span> <span class="st">"healthy"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dysbiosis_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dysbiosisMedianCLV.html">dysbiosisMedianCLV</a></span><span class="op">(</span><span class="va">WirbelJ_2018</span>,</span>
<span>                                  dist_mat <span class="op">=</span> <span class="va">dist.mat</span>,</span>
<span>                                  reference_samples <span class="op">=</span> <span class="va">ref.samples</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dysbiosis_1</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">disease</span>,<span class="va">score</span>, color<span class="op">=</span><span class="va">disease</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.3</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.1</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>CRC<span class="op">=</span><span class="st">"brown3"</span>, healthy <span class="op">=</span><span class="st">"steelblue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">""</span>, y<span class="op">=</span><span class="st">"Dysbiosis Score"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Introduction_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="cloud-based-locally-linear-unbiased-dysbiosis-cloud-test">Cloud-based LOcally linear Unbiased Dysbiosis (CLOUD) test<a class="anchor" aria-label="anchor" href="#cloud-based-locally-linear-unbiased-dysbiosis-cloud-test"></a>
</h3>
<p>Cloud-based LOcally linear Unbiased Dysbiosis (CLOUD) test is a
non-parametric test and returns a measure of dysbiosis. The function was
adapted from the original article by <a href="https://doi.org/10.1186/s40168-018-0514-4" class="external-link">Montassier E et
al. 2018</a>. Here, a user defines a set of reference samples from which
distance of every other sample is calculated. An important aspect to
note in this function is that the method is slightly modified to
accommodate equal number of samples in reference and test group. When
calculating the stats for reference samples the number of neighbors is
defined as number of reference samples minus 1. For test samples, IF
number of test samples matches number of samples in reference group,
then number of neighbors is also defined as number of reference samples
minus 1. This is done to avoid returning NAs in outputs while retaining
maximum number of reference samples.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">cloud.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cloudStatistic.html">cloudStatistic</a></span><span class="op">(</span><span class="va">WirbelJ_2018</span>,</span>
<span>                                dist_mat <span class="op">=</span> <span class="va">dist.mat</span>,</span>
<span>                                reference_samples <span class="op">=</span> <span class="va">ref.samples</span>,</span>
<span>                                ndim<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">cloud.results</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                    stats    pvals            ids   log2Stats   study_name</span></span>
<span><span class="co">#&gt; MMPU29365221ST 1.0186169 0.343750 MMPU29365221ST  0.02661160 WirbelJ_2018</span></span>
<span><span class="co">#&gt; MMPU68403337ST 0.9591212 0.625000 MMPU68403337ST -0.06021501 WirbelJ_2018</span></span>
<span><span class="co">#&gt; MMPU72854103ST 0.9914070 0.484375 MMPU72854103ST -0.01245058 WirbelJ_2018</span></span></code></pre></div>
<p>Here, the <code>log2Stats</code> can be used as a dysbiosis
score.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cloud.results</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">disease</span>, <span class="va">log2Stats</span>, color<span class="op">=</span><span class="va">disease</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.3</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.1</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>CRC<span class="op">=</span><span class="st">"brown3"</span>, healthy <span class="op">=</span><span class="st">"steelblue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">""</span>, y<span class="op">=</span><span class="st">"Dysbiosis CLOUD Score"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 60 rows containing non-finite values (stat_boxplot).</span></span>
<span><span class="co">#&gt; Warning: Removed 60 rows containing missing values (geom_point).</span></span></code></pre></div>
<p><img src="Introduction_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="dysbiosis-score-based-on-euclidean-distance-to-group-centroids">Dysbiosis score based on Euclidean distance to group centroids<a class="anchor" aria-label="anchor" href="#dysbiosis-score-based-on-euclidean-distance-to-group-centroids"></a>
</h3>
<p>Calculates difference in <code>euclidean distance (ED)</code> for a
sample to group centroids. For example, sample_1 to control centroid
minus sample_1 to case centroid. The user can provide a custom distance
matrix. This approach was used in <a href="https://doi.org/10.1093/femsec/fix136" class="external-link">AlShawaqfeh MK et
al. (2017)</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dysbiosis_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/euclideanDistCentroids.html">euclideanDistCentroids</a></span><span class="op">(</span><span class="va">WirbelJ_2018</span>,</span>
<span>                                      dist_mat <span class="op">=</span> <span class="va">dist.mat</span>,</span>
<span>                                      use_squared <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                      group_col <span class="op">=</span> <span class="st">"disease"</span>,</span>
<span>                                      control_label <span class="op">=</span> <span class="st">"healthy"</span>,</span>
<span>                                      case_label <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span></span>
<span><span class="va">dysbiosis_2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                     CentroidDist_CRC CentroidDist_healthy CentroidDist_score</span></span>
<span><span class="co">#&gt; CCMD11006829ST-21-0        0.2912347            0.3684569         0.07722225</span></span>
<span><span class="co">#&gt; CCMD12232071ST-21-0        0.2744104            0.3415175         0.06710710</span></span>
<span><span class="co">#&gt; CCMD13071240ST-21-0        0.2172320            0.2697302         0.05249821</span></span></code></pre></div>
<p>Here, the <code>CentroidDist_score</code> can be used as a dysbiosis
score.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dysbiosis_2</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">disease</span>, <span class="va">CentroidDist_score</span>, color<span class="op">=</span><span class="va">disease</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.3</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.1</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>CRC<span class="op">=</span><span class="st">"brown3"</span>, healthy <span class="op">=</span><span class="st">"steelblue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">""</span>, y<span class="op">=</span><span class="st">"Dysbiosis CLOUD Score"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Introduction_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19044)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] dysbiosisR_0.1.0  microbiome_1.18.0 ggplot2_3.3.6     phyloseq_1.40.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] nlme_3.1-157           bitops_1.0-7           fs_1.5.2              </span></span>
<span><span class="co">#&gt;  [4] usedist_0.4.0          rprojroot_2.0.3        GenomeInfoDb_1.32.2   </span></span>
<span><span class="co">#&gt;  [7] tools_4.2.1            bslib_0.4.0            utf8_1.2.2            </span></span>
<span><span class="co">#&gt; [10] R6_2.5.1               vegan_2.6-2            DBI_1.1.3             </span></span>
<span><span class="co">#&gt; [13] BiocGenerics_0.42.0    mgcv_1.8-40            colorspace_2.0-3      </span></span>
<span><span class="co">#&gt; [16] permute_0.9-7          rhdf5filters_1.8.0     ade4_1.7-19           </span></span>
<span><span class="co">#&gt; [19] withr_2.5.0            tidyselect_1.1.2       compiler_4.2.1        </span></span>
<span><span class="co">#&gt; [22] textshaping_0.3.6      cli_3.3.0              Biobase_2.56.0        </span></span>
<span><span class="co">#&gt; [25] desc_1.4.2             labeling_0.4.2         sass_0.4.2            </span></span>
<span><span class="co">#&gt; [28] scales_1.2.1           pkgdown_2.0.6          systemfonts_1.0.4     </span></span>
<span><span class="co">#&gt; [31] stringr_1.4.1          digest_0.6.29          rmarkdown_2.16        </span></span>
<span><span class="co">#&gt; [34] XVector_0.36.0         pkgconfig_2.0.3        htmltools_0.5.3       </span></span>
<span><span class="co">#&gt; [37] highr_0.9              fastmap_1.1.0          rlang_1.0.5           </span></span>
<span><span class="co">#&gt; [40] rstudioapi_0.14        farver_2.1.1           jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt; [43] generics_0.1.3         jsonlite_1.8.0         dplyr_1.0.9           </span></span>
<span><span class="co">#&gt; [46] RCurl_1.98-1.6         magrittr_2.0.3         GenomeInfoDbData_1.2.8</span></span>
<span><span class="co">#&gt; [49] biomformat_1.24.0      Matrix_1.5-1           Rcpp_1.0.8.3          </span></span>
<span><span class="co">#&gt; [52] munsell_0.5.0          S4Vectors_0.34.0       Rhdf5lib_1.18.2       </span></span>
<span><span class="co">#&gt; [55] fansi_1.0.3            ape_5.6-2              lifecycle_1.0.2       </span></span>
<span><span class="co">#&gt; [58] stringi_1.7.6          yaml_2.3.5             MASS_7.3-57           </span></span>
<span><span class="co">#&gt; [61] zlibbioc_1.42.0        rhdf5_2.40.0           Rtsne_0.16            </span></span>
<span><span class="co">#&gt; [64] plyr_1.8.7             grid_4.2.1             parallel_4.2.1        </span></span>
<span><span class="co">#&gt; [67] crayon_1.5.1           lattice_0.20-45        Biostrings_2.64.0     </span></span>
<span><span class="co">#&gt; [70] splines_4.2.1          multtest_2.52.0        knitr_1.40            </span></span>
<span><span class="co">#&gt; [73] pillar_1.8.1           igraph_1.3.1           reshape2_1.4.4        </span></span>
<span><span class="co">#&gt; [76] codetools_0.2-18       stats4_4.2.1           glue_1.6.2            </span></span>
<span><span class="co">#&gt; [79] evaluate_0.16          data.table_1.14.2      vctrs_0.4.1           </span></span>
<span><span class="co">#&gt; [82] foreach_1.5.2          gtable_0.3.1           purrr_0.3.4           </span></span>
<span><span class="co">#&gt; [85] tidyr_1.2.0            assertthat_0.2.1       cachem_1.0.6          </span></span>
<span><span class="co">#&gt; [88] xfun_0.31              ragg_1.2.2             survival_3.3-1        </span></span>
<span><span class="co">#&gt; [91] tibble_3.1.7           iterators_1.0.14       memoise_2.0.1         </span></span>
<span><span class="co">#&gt; [94] IRanges_2.30.0         cluster_2.1.3          ellipsis_0.3.2</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Sudarshan Shetty.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
